engine-tests/src/test_utils/uniswap.rs

```diff
@@ -335,6 +345,42 @@ impl SwapRouter {
             data,
         }
     }
+
+    pub fn exact_input(&self, params: ExactInputParams, nonce: U256) -> TransactionLegacy {
+        let path: Vec<u8> = {
+            // The encoding here is 32-byte address, then 3-byte (24-bit) fee, alternating
+            let mut result = Vec::with_capacity(32 + 35 * params.path.len());
+            result.extend_from_slice(params.token_in.as_bytes());
+            for (fee, token) in params.path.iter() {
+                let fee_bytes = fee.to_be_bytes();
+                result.extend_from_slice(&fee_bytes[5..8]);
+                result.extend_from_slice(token.as_bytes());
+            }
+            result
+        };
+        let data = self
+            .0
+            .abi
+            .function("exactInput")
+            .unwrap()
+            .encode_input(&[ethabi::Token::Tuple(vec![
+                ethabi::Token::Bytes(path),
+                ethabi::Token::Address(params.recipient.raw()),
+                ethabi::Token::Uint(params.deadline),
+                ethabi::Token::Uint(params.amount_in),
+                ethabi::Token::Uint(params.amount_out_min),
+            ])])
+            .unwrap();
+
+        TransactionLegacy {
+            nonce,
+            gas_price: Default::default(),
```

Comment:
Another nit, personally I don't like `Default::default()` auto magic but, again its just a test, so not too worried if this gets changed or not.